plugin/trino-ldap-group-provider/src/test/java/io/trino/plugin/ldapgroup/TestLdapGroupProviderIntegration.java

```diff
@@ -0,0 +1,314 @@
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+public class TestLdapGroupProviderIntegration
+{
+    private final LdapGroupProviderFactory factory = new LdapGroupProviderFactory();
+
+    private final Closer closer;
+
+    private final Map<String, String> baseConfig;
+
+    private final DisposableSubContext clients;
+    private final DisposableSubContext developers;
+    private final DisposableSubContext qa;
+
+    public TestLdapGroupProviderIntegration()
+            throws Exception
+    {
+        closer = Closer.create();
+        Network network = Network.newNetwork();
+        closer.register(network::close);
+
+        TestingOpenLdapServer openLdapServer = closer.register(new TestingOpenLdapServer(network));
+        openLdapServer.start();
+
+        baseConfig = Map.of(
+                "ldap.url", openLdapServer.getLdapUrl(),
+                "ldap.allow-insecure", "true",
+                "ldap.admin-user", "cn=admin,dc=trino,dc=testldap,dc=com",
+                "ldap.admin-password", "admin",
+                "ldap.user-base-dn", "ou=users,dc=trino,dc=testldap,dc=com",
+                "ldap.user-search-filter", "(cn={0})");
+
+        DisposableSubContext groupsOU = openLdapServer.createOrganization("groups");
+        DisposableSubContext externalGroupsOU = openLdapServer.createOrganization("external", groupsOU);
+        DisposableSubContext usersOU = openLdapServer.createOrganization("users");
+
+        DisposableSubContext johnb = openLdapServer.createUser(usersOU, "johnb", ImmutableMap.of(
+                "employeeNumber", "1000",
+                "mail", "johnb@company.com"));
+        DisposableSubContext alicea = openLdapServer.createUser(usersOU, "alicea", ImmutableMap.of(
+                "employeeNumber", "1001",
+                "mail", "alicea@trino.io",
+                "givenname", "Alice"));
+        DisposableSubContext bobq = openLdapServer.createUser(usersOU, "bobq", ImmutableMap.of(
+                "employeeNumber", "1002"));
+        openLdapServer.createUser(usersOU, "carlp", ImmutableMap.of(
+                "employeeNumber", "1003"));
+
+        clients = openLdapServer.createGroup(externalGroupsOU, "clients");
+        openLdapServer.addUserToGroup(johnb, clients);
+        openLdapServer.addUserToGroup(alicea, clients);
+
+        developers = openLdapServer.createGroup(groupsOU, "developers");
+        openLdapServer.addUserToGroup(alicea, developers);
+
+        qa = openLdapServer.createGroup(groupsOU, "qa");
+        openLdapServer.addUserToGroup(alicea, qa);
+        openLdapServer.addUserToGroup(bobq, qa);
+    }
+
+    @AfterAll
+    public void close()
+            throws Exception
+    {
+        closer.close();
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideUsersAndExpectedGroups")
+    public void testGetGroups(ConfigBuilder configBuilder, String userName, Set<String> expectedGroups)
```

Comment:
It is not generally recommended to use `ParameterizedTest` - In general, they make it harder to debug tests - We could try to create some sort of an `assertYYY` - For reference we could try something like `TestDecimalCoercers`

Reply:
will look into this a bit, but I personally don't see an overall ban on `ParametrizedTests` as a good thing. they may be overused but they do have a place